### PR TITLE
Add type casting to postgres queries

### PR DESCRIPTION
**Pull request by krynble**
This PR extends functionality created on https://github.com/n8n-io/n8n/pull/1298 by @mutdmour 

Types were added to both Insert and Update operations. The implementation makes types optional and is backward compatible.

Also tested on QuestDB, TimescaleDB and Cratedb.

Lastly, added description and placeholder tips for users to show how to utilize.

### CODE REVIEW

File: packages/nodes-base/nodes/Postgres/Postgres.node.functions.ts [hunk 1]
```diff
@@ -68,21 +68,26 @@ export async function pgInsert(
...
+	const columnSet = new pgp.helpers.ColumnSet(columns);
 
 	// Generate the multi-row insert query and return the id of new row
 	returnFields = returnFields.map(value => value.trim()).filter(value => !!value);
 	const query =
-		pgp.helpers.insert(insertItems, cs, te) +
+		pgp.helpers.insert(insertItems, columnSet, te) +
 		(returnFields.length ? ` RETURNING ${returnFields.join(',')}` : '');
 
+	console.log(query);
```
Comment (BHesseldieck) [inline packages/nodes-base/nodes/Postgres/Postgres.node.functions.ts:89]: leftover
```suggestion
```

File: packages/nodes-base/nodes/Postgres/Postgres.node.functions.ts [hunk 2]
```diff
@@ -109,21 +114,36 @@ export async function pgUpdate(
 	const updateKey = getNodeParam('updateKey', 0) as string;
 	const columnString = getNodeParam('columns', 0) as string;
 
-	const columns = columnString.split(',').map(column => column.trim());
+	const updateColumnParts = updateKey.split(':');
+	const updateColumn = {
+		name: updateColumnParts[0],
+		cast: updateColumnParts[1],
+	};
```
Comment (BHesseldieck) [inline packages/nodes-base/nodes/Postgres/Postgres.node.functions.ts:121]: I think a destruction makes it more readable here 😊 
```suggestion
	const [updateColumnName, updateColumnCast] = updateKey.split(':');
	const updateColumn = {
		name: updateColumnName,
		cast: updateColumnCast,
	};
```

File: packages/nodes-base/nodes/QuestDb/QuestDb.node.ts [hunk 3]
```diff
@@ -107,22 +107,6 @@ export class QuestDb implements INodeType {
...
-						operation: [
-							'insert',
-						],
-					},
-				},
-				default: '',
-				placeholder: 'id,name,description',
-				description:
-					'Comma separated list of the properties which should used as columns for the new rows.',
-			},
```
Comment (BHesseldieck) [inline packages/nodes-base/nodes/QuestDb/QuestDb.node.ts:125]: Does QuestDB not support column type casting?

File: packages/nodes-base/test/nodes/Postgres/Postgres.node.functions.test.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
+const PostgresFun = require('../../../nodes/Postgres/Postgres.node.functions')
+const pgPromise = require('pg-promise');
+
+describe('pgUpdate', () => {
```
Comment (BHesseldieck) [inline packages/nodes-base/test/nodes/Postgres/Postgres.node.functions.test.js:4]: love that we are adding tests, however are just happy paths covered here. Don't know if adding more tests is out of scope of this PR

**Review by BHesseldieck — APPROVED** [review]

File: packages/nodes-base/nodes/QuestDb/QuestDb.node.ts (hunk 3 above)
Comment (krynble) [reply packages/nodes-base/nodes/QuestDb/QuestDb.node.ts:125]: It kinda does, but actually this column list was not being used. The implementation gets the json information passed in and uses json keys as columns so I removed this option as it's useless.

File: packages/nodes-base/test/nodes/Postgres/Postgres.node.functions.test.js (hunk 4 above)
Comment (krynble) [reply packages/nodes-base/test/nodes/Postgres/Postgres.node.functions.test.js:4]: I believe the tests cover the extend of this PR as we're now adding functionality to existing node and this is being covered.

Good that you mentioned this. I added tests to the insert operation that I incremented. It was missing.

Regarding non-happy path, actually the functions don't have much to fail. Only issues that might arise are query issues which would throw regular Errors and would be catched by the node and displayed to our user.

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with n8n@0.114.0

**Pull request merged**